syncd/syncd_flex_counter.cpp

```diff
@@ -548,13 +587,16 @@ void FlexCounter::removeCounterPlugin(
 
     FlexCounter &fc = getInstance(instanceId);
 
+    std::unique_lock<std::mutex> lkMgr(fc.m_mtx);
+
     fc.m_queuePlugins.clear();
     fc.m_portPlugins.clear();
     fc.m_priorityGroupPlugins.clear();
 
     // Remove flex counter if all counter IDs and plugins are unregistered
     if (fc.isEmpty())
     {
+        lkMgr.unlock();
```

Comment:
please don't do manual unlocking, this is really bad design, use scope lock/unlock